### PR TITLE
BUG: CategoricalIndex.format

### LINKED ISSUE #35439: BUG: output of df.to_string depends on whether columns is a CategoricalIndex or not
The output of ``df.to_string`` is different if the columns is a ``CategoricalIndex`` rather than a normal ``Index``:

```python
>>> data = [[4, 2], [3, 2], [4, 3]]
>>> cols = ["aaaaaaaaa", "b"]
>>>  pd.DataFrame(data, columns=cols)
   aaaaaaaaa  b
0          4  2
1          3  2
2          4  3
>>> pd.DataFrame(data, columns=pd.CategoricalIndex(cols))
   aaaaaaaaa  b
0          4          2
1          3          2
2          4          3
```

We can see that the width of the "b" column depends on the width of the "a" column, if the columns are a ``CategoricalIndex``.

These two examples should return the same repr.

### PR DESCRIPTION
**Pull request by topper-123**
- [x] closes #35439
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I've temporarily put the whatsnewentry in the v.1.1.0 release note, because there isn't a v.1.1.1 version yet. I'll move it, before this is merged.

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @topper-123 for the PR.

The regression was caused by #35118. Categorical types other than object were also affected. maybe need to parameterise test with other values for `cols`

```
>>> pd.__version__
'1.2.0.dev0+10.g3b1d4f1ee'
>>>
>>> data = [[4, 2], [3, 2], [4, 3]]
>>> cols = [1, None]
>>> res = pd.DataFrame(data, columns=cols)
>>> print(res)
   1  NaN
0  4    2
1  3    2
2  4    3
>>>
>>> res = pd.DataFrame(data, columns=pd.CategoricalIndex(cols))
>>> print(res)
   1    NaN
0    4    2
1    3    2
2    4    3
>>>
```

```
>>> pd.__version__
'1.0.5'
>>>
>>> data = [[4, 2], [3, 2], [4, 3]]
>>> cols = [1, None]
>>> res = pd.DataFrame(data, columns=cols)
>>> print(res)
   1  NaN
0  4    2
1  3    2
2  4    3
>>>
>>> res = pd.DataFrame(data, columns=pd.CategoricalIndex(cols))
>>> print(res)
   1.0  NaN
0    4    2
1    3    2
2    4    3
>>>
```

> I've temporarily put the whatsnewentry in the v.1.1.0 release note, because there isn't a v.1.1.1 version yet.

doc\source\whatsnew\v1.1.1.rst now merged to master

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -197,9 +197,6 @@ def _format_data(self, name=None):
         # we are formatting thru the attributes
         return None
 
-    def _format_with_header(self, header, na_rep="NaN") -> List[str]:
-        return header + [pprint_thing(x) for x in self._range]
-
```
Comment (topper-123) [inline pandas/core/indexes/range.py:202]: The added tests revealed that this made the output from ``.format`` different than for ``Int64Index``:

```python
>>> pd.RangeIndex(0, 18, 2).format()
['0', '2', '4', '6', '8', '10', '12', '14', '16']
>>> pd.Int64Index(range(0, 18, 2)).format()
 ['0 ', '2 ', '4 ', '6 ', '8 ', '10', '12', '14', '16']
```
 Notice the extra space for one-digit scalars in the ``Int64Index`` case. The output are identical after merging this PR.

**Comment by topper-123** [comment]
Updated.

**Comment by jreback** [comment]
thanks @topper-123 very nice!

**Comment by simonjayhawkins** [comment]
@MeeseeksDev backport to 1.1.x

**Pull request merged**